src/Nest/CommonAbstractions/Infer/Field/FieldExpressionVisitor.cs

```diff
@@ -14,7 +14,9 @@ namespace Nest
 {
 	internal class HasConstantExpressionVisitor : ExpressionVisitor
 	{
-		public bool Found { get; private set; }
+		public bool Found => this.FoundConstants.Any(b => b);
```

Comment:
You're right, i wonder if we can just make it a bool flag that once set is never unset? Would safe on allocs too.